test/perms.js

```diff
@@ -0,0 +1,148 @@
+/* jslint node: true */
+/* global describe: false, before: false, after: false, it: false, afterEach: false, beforeEach: false */
+/* jshint -W030 */
+'use strict';
+
+var NATS = require ('../'),
+nsc = require('./support/nats_server_control'),
+ncu = require('./support/nats_conf_utils'),
+os = require('os'),
+fs = require('fs'),
+path = require('path'),
+should = require('should'),
+nuid = require('nuid');
+
+
+describe('Auth Basics', function() {
+
+    var PORT = 6758;
+    var server;
+
+    // Start up our own nats-server
+    before(function (done) {
+        var conf = {
+            authorization: {
+                ADMIN: {
+                    publish: ">",
+                    subscribe: ">"
+                },
+                SUB: {
+                    subscribe: "bar",
+                    publish: "bar",
+                },
+                PUB: {
+                    subscribe: "foo",
+                    publish: "foo"
+                },
+                users: [
+                    {user: 'admin', password: 'admin', permission: '$ADMIN'},
+                    {user: 'foo', password: 'foo', permission: '$PUB'},
+                    {user: 'bar', password: 'bar', permission: '$SUB'}
+                ]
+            }
+        };
+        var cf = path.resolve(os.tmpdir(), 'conf-' + nuid.next() +'.conf');
+        ncu.writeFile(cf, ncu.JsonToYaml(conf));
+        server = nsc.start_server(PORT, ['-c', cf], done);
+    });
+
+    // Shutdown our server
+    after(function () {
+        nsc.stop_server(server);
+    });
+
+    it('foo can subscribe/pub foo', function (done) {
```

Comment:
I would not test the positive cases here, this is a server test, not a client. You want to test a part of the code that handles the failure, so I would keep only negative tests where we try to pub/sub and get an error and check that connection is not closed.